docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/private-api-is-used-on-the-project-level.md

```diff
@@ -452,30 +527,41 @@ class CustomerAccessFilter implements CustomCustomerAccessFilter
 }
 ```
 
-Example of related error in the Evaluator output:
-
-```bash
-------------------------------------------------------------------------------------------------------------------------
-Please avoid usage of Spryker\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessReaderInterface in Pyz\Zed\CustomerAccess\Business\CustomerAccess\CustomerAccessFilter
-------------------------------------------------------------------------------------------------------------------------
-```
-
-Example of resolving the error by ?
+### Resolving the error
 
-The solution for the error is the same as in `Resolving the error`[PrivateApi:Facade](#privateapi-facade)
+To resolve the error provided in the example, try the following in the provided order:
+1. Recommended: Extend the functionality using the [Configuration strategy](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#configuration).
+2. Recommended: Extend the functionality using the [Plug and Play strategy](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#plug-and-play).
+3. Recommended: Extend the functionality using the [Project Modules strategy](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#project-modules).
+4. Not recommended: Copy the functionality from the core to the project level:
+    1. Register the missing extension point in [Spryker Ideas](https://spryker.ideas.aha.io/), so we add it in the future.
+    2. Copy private API core entities to the project level and give them unique names.
+    3. As soon as the extension point in core is released, refactor the code added in step 4.2 using the strategies in steps 1-3.
+       While it's not refactored, auto-upgrades are not supported, and the effort to update the project may be bigger and require more manual work.
 
+---
```

Comment:
Please provide an example of solving the issue

Reply:
done